whois-api/src/main/java/net/ripe/db/whois/api/rest/HealthCheckService.java

```diff
@@ -1,31 +1,49 @@
 package net.ripe.db.whois.api.rest;
 
 import net.ripe.db.whois.common.iptree.IpTreeCacheManager;
+import net.ripe.db.whois.common.jmx.JmxBase;
 import net.ripe.db.whois.common.source.SourceContext;
 import org.apache.commons.io.FileUtils;
 import org.apache.commons.lang3.StringUtils;
+import org.glassfish.jersey.client.ClientProperties;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.beans.factory.annotation.Value;
 import org.springframework.dao.DataAccessException;
 import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jmx.export.annotation.ManagedAttribute;
+import org.springframework.jmx.export.annotation.ManagedOperation;
+import org.springframework.jmx.export.annotation.ManagedOperationParameter;
+import org.springframework.jmx.export.annotation.ManagedResource;
 import org.springframework.scheduling.annotation.Scheduled;
 import org.springframework.stereotype.Component;
 
 import javax.sql.DataSource;
 import javax.ws.rs.GET;
 import javax.ws.rs.Path;
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
 import javax.ws.rs.core.Response;
 import javax.ws.rs.core.Response.Status;
 import java.io.File;
 import java.io.IOException;
 import java.util.concurrent.atomic.AtomicBoolean;
 
 @Component
+@ManagedResource(objectName = JmxBase.OBJECT_NAME_BASE + "Loadbalancer", description = "Loadbalancer switch")
+
+/**
+ * in jmxterm, run with:
+ *      bean net.ripe.db.whois:name=Loadbalancer
+ *      run setLoadbalancerEnabled true|false
+ *
+ */
+
+
 @Path("/healthcheck")
-public class HealthCheckService {
+public class HealthCheckService extends JmxBase {
```

Comment:
This class is getting increasingly complex, is it time to spearate each ```HealthCheck``` using an interface?

Reply:
Wanted to keep the changes as minimal as possible. That is, just make the controller a managed bean. But I agree the individual health checks should probably move to their own classes. I'll update

Reply:
I'm OK with leaving as-is for now also 👍

Reply:
On another thought, I will add another class, which will keep the state whether to enable or disable from the LB and make this managed by JMX instead of the controller. This way it can also be usable from whois-internal